File: aula8/exercicio_3/java_logic/Exercicio3Modular.java
// Exercício 3 - Validador de Senha
import java.util.Scanner;

public class Exercicio3Modular {

    public static boolean validarSenha(String senha) {
        // O retorno booleano facilita muito usar a função dentro do while
        return senha.length() >= 8;
    }

    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);

        String senha = entrada.nextLine();

        System.out.print("Digite uma senha: ");
        senha = entrada.nextLine();

        while (!validarSenha(senha)) {
            System.out.println("Senha invalida. Digite pelo menos 8 caracteres.");
            System.out.print("Digite uma senha: ");
            senha = entrada.nextLine();
        }

        System.out.println("Senha valida.");

        entrada.close();
    }
}

File: aula8/exercicio_3/cpp_logic/exercicio3_modular.cpp

#include <iostream>
#include <string>
using namespace std;

bool validarSenha(string senha) {
    // Em C++, bool também encaixa muito bem na lógica de repetição
    return senha.length() >= 8;
}

int main() {
    string senha;


    cout << "Digite uma senha: ";
    getline(cin, senha);

    while (!validarSenha(senha)) {
        cout << "Senha invalida. Digite pelo menos 8 caracteres." << endl;
        cout << "Digite uma senha: ";
        getline(cin, senha);
    }

    cout << "Senha valida." << endl;

    return 0;
}
